addons/l10n_in_edi_ewaybill/models/account_move.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # Transaction Details
+    l10n_in_transaction_type = fields.Selection(
+        [
+            ("1", "Regular"),
+            ("2", "Bill To-Ship To"),
+            ("3", "Bill From-Dispatch From"),
+            ("4", "Combination of 2 and 3"),
+        ],
+        string="Transaction Type (E-waybill)",
+        tracking=True,
+    )
+    l10n_in_type_id = fields.Many2one("l10n.in.ewaybill.type", "Document Type", tracking=True)
+    l10n_in_subtype_id = fields.Many2one("l10n.in.ewaybill.type", "Sub Supply Type", tracking=True)
+
+    # transportation details
+    l10n_in_distance = fields.Integer("Distnace", tracking=True)
+    l10n_in_mode = fields.Selection([
+        ("0", "Managed by Transporter"),
+        ("1", "By Road"),
+        ("2", "Rail"),
+        ("3", "Air"),
+        ("4", "Ship")],
+        compute="_compute_l10n_in_mode", store=True, readonly=False,
+        string="Transportation Mode", copy=False, tracking=True)
+
+    # Vehicle Number and Type required when transportation mode is By Road.
+    l10n_in_vehicle_no = fields.Char("Vehicle Number", copy=False, tracking=True)
```

Comment:
why tracking?  How many times are you going to change this?

Reply:
it's possible that after Creating E-waybill this detail is changed.
normally when the truck is breakdown, accident or something like that.

FYI:  audit trail (edit log) mandated by the Government of India.